packages/@coorpacademy-components/src/atom/button-link/index.tsx

```diff
@@ -2,13 +2,46 @@ import React, {useCallback} from 'react';
 import {noop} from 'lodash/fp';
 import classnames from 'classnames';
 import Link from '../link';
+import FaIcon from '../icon';
 import {ICONS} from '../../util/button-icons';
 import propTypes, {ButtonLinkProps, IconType} from './types';
 import style from './style.css';
 
 const getButtonContent = (icon?: IconType, label?: string) => {
-  const {type, position} = icon || {type: '', position: ''};
+  const {type, faIconName, faIconColor, faIconBackgroundColor, faIconSize, position, customStyle} =
```

Comment:
```suggestion
  const {type, faIconProps, position, customStyle} =
```

Reply:
```suggestion
  const {type, faIcon, position, customStyle} =
```
`{name, color, backgroundColor, size} = faIcon`

(is customStyle still necessary with the faIcon approach)

Reply:
Thank you @youssefezzahi96 @AdrieanKhisbe , I've updated